### PR TITLE
[DUOS-1660][risk=no] Require library card for creating a DAR

### PR DESCRIPTION
**Pull request by rushtong**
## Addresses
https://broadworkbench.atlassian.net/browse/DUOS-1660

Adds a check to the create DAR endpoint to ensure that the create user has at least one library card.

Manually tested with a user that does have a library card and one that does not. There is no change to Library Card users' ability to submit DARs. Non-LC users will see a 403 error and be redirected back to their console. The UI work for handling this case is covered in https://broadworkbench.atlassian.net/browse/DUOS-1657

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/main/java/org/broadinstitute/consent/http/resources/DataAccessRequestResourceVersion2.java [hunk 1]
```diff
@@ -92,6 +93,12 @@ public Response createDataAccessRequest(
       @Auth AuthUser authUser, @Context UriInfo info, String dar) {
     try {
       User user = findUserByEmail(authUser.getEmail());
+      if (Objects.isNull(user.getLibraryCards()) || user.getLibraryCards().isEmpty()) {
+        return Response
+          .status(Response.Status.FORBIDDEN)
```
Comment (quazi-h) [inline src/main/java/org/broadinstitute/consent/http/resources/DataAccessRequestResourceVersion2.java:98]: Is this the piece that would trigger the 403 response in `./paths/darV2.yaml`?

**Review by quazi-h — APPROVED** [review]
Changes to check for a library card during DAR submission looks correct to me. This should return a 403 response from the API when attempting to submit a DAR with a user without a library card.

**Review by JVThomas — APPROVED** [review]
Tested locally and confirmed that the lc checks are working as expected. Code looks good and tests have been updated to reflect the changes.

File: src/main/java/org/broadinstitute/consent/http/resources/DataAccessRequestResourceVersion2.java (hunk 1 above)
Comment (rushtong) [reply src/main/java/org/broadinstitute/consent/http/resources/DataAccessRequestResourceVersion2.java:98]: Yes, that's correct.

**Pull request merged**